fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostCalculator.java

```diff
@@ -158,22 +192,10 @@ public CostEstimate visitPhysicalNestedLoopJoin(
 
             StatsDeriveResult leftStatistics = context.getChildStatistics(0);
             StatsDeriveResult rightStatistics = context.getChildStatistics(1);
-            List<Id> leftIds = context.getChildOutputIds(0);
-            List<Id> rightIds = context.getChildOutputIds(1);
-
-            // TODO: handle some case
-            // handle cross join, onClause is empty .....
-            if (nestedLoopJoin.getJoinType().isCrossJoin()) {
-                return new CostEstimate(
-                        leftStatistics.computeColumnSize(leftIds) + rightStatistics.computeColumnSize(rightIds),
-                        rightStatistics.computeColumnSize(rightIds),
-                        0);
-            }
 
-            // TODO: network 0?
             return new CostEstimate(
-                    (leftStatistics.computeColumnSize(leftIds) + rightStatistics.computeColumnSize(rightIds)) / 2,
-                    rightStatistics.computeColumnSize(rightIds),
+                    leftStatistics.computeSize() * rightStatistics.computeSize(),
+                    rightStatistics.computeSize(),
```

Comment:
In my understanding, the memory is the size of two blocks, from left child and right child respectively.

Reply:
This PR aim to get a correct plan of colocate join and bucket shuffle join. Good cost model is not including. We need to adjust it finely after this PR and https://github.com/apache/doris/pull/12767